src/rdf4cpp/IRIFactory.hpp

```diff
@@ -70,6 +70,15 @@ struct IRIFactory {
      * @param prefix
      * @param expanded
      */
+    [[nodiscard]] IRIFactoryError assign_prefix_checked(std::string_view prefix, std::string_view expanded);
```

Comment:
Naming should be the other way around `assign_prefix` and `assign_prefix_unchecked`, this is the safer approach